### PR TITLE
E-1197 feat(ui): sorting support for tables

### PR DESCRIPTION
**Pull request by ifiri**

### CODE REVIEW

**Comment by ifiri** [comment]
@dudiq

File: packages/ui/src/kit/VTable/VTableHeadCell/VTableHeadCell.vue [hunk 1]
```diff
@@ -1,21 +1,97 @@
 <template>
   <th :class="vTableHeadCellCssClass">
-    <slot />
+    <a
```
Comment (dudiq) [inline packages/ui/src/kit/VTable/VTableHeadCell/VTableHeadCell.vue:3]: if we are not using `<a>` for other cases, we can use `v-if=isSortable` for `<a>` tag here

File: packages/ui/src/kit/VTable/VTableHeadCell/VTableHeadCell.vue [hunk 2]
```diff
@@ -1,21 +1,97 @@
...
+const SORTING_DIRECTIONS = {
+  DESC: 'desc',
+  ASC: 'asc',
+};
 
 export default {
   name: 'VTableHeadCell',
+
+  props: {
+    sortingBy: {
```
Comment (dudiq) [inline packages/ui/src/kit/VTable/VTableHeadCell/VTableHeadCell.vue:33]: no need to pass `sortingBy`, it's enought to pass only sort direction

File: packages/ui/src/kit/VTable/VTableHeadCell/VTableHeadCell.vue [hunk 3]
```diff
@@ -1,21 +1,97 @@
...
+      type: String,
+      default: null,
+    },
+
+    sortingValue: {
```
Comment (dudiq) [inline packages/ui/src/kit/VTable/VTableHeadCell/VTableHeadCell.vue:38]: and no need to pass `sortingValue` to.
head cell does not know about any values, only sort direction

File: packages/ui/src/kit/VTable/VTableHeadCell/VTableHeadCell.vue (hunk 2 above)
Comment (ifiri) [reply packages/ui/src/kit/VTable/VTableHeadCell/VTableHeadCell.vue:33]: you aren't right
we anyway should pass some flag, and this is not direction
so you just suggest to replace `sortingBy` to `isSortable`
also if cell knows about sort direction, cell anyways know about direction and responsible for that, so no crime to let it know about their sorting value just to determine that current cell is sortable

for now I will pass `isSortable` flag and this change will make using sortable table more difficult

Comment (dudiq) [reply packages/ui/src/kit/VTable/VTableHeadCell/VTableHeadCell.vue:33]: direction can be in those values:
- no direction (default state)
- asc
- desc

and when you click to head cell, you switching only this values and emit event about new direction

**Comment by dudiq** [comment]
approved

**Comment by kazurus** [comment]
approved

**Pull request merged**